stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/plugin/state/internal/ChatClientStateCalls.kt

```diff
@@ -27,18 +27,29 @@ import io.getstream.chat.android.offline.plugin.state.channel.ChannelState
 import io.getstream.chat.android.offline.plugin.state.channel.thread.ThreadState
 import io.getstream.chat.android.offline.plugin.state.querychannels.QueryChannelsState
 import kotlinx.coroutines.CoroutineScope
+import java.util.Date
+
+private const val MAX_TIME_OF_STATE_MILI = 300
 
 /**
  * Adapter for [ChatClient] that wraps some of it's request.
  */
-internal class ChatClientStateCalls(
+internal class ChatClientStateCalls internal constructor(
     private val chatClient: ChatClient,
     private val state: StateRegistry,
-    private val scope: CoroutineScope
+    private val scope: CoroutineScope,
 ) {
+
+    private val requestTimeMap: MutableMap<Int, Date> = mutableMapOf()
```

Comment:
Since `ChatClientStateCalls` seems to be a singleton, don't you think we should have some mechanism to clear `requestTimeMap` map?

Reply:
We do. I'll add that